src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderServiceTest.java

```diff
@@ -288,27 +284,27 @@ void getRollupCountForCriteriaGroupsGeneratesAndRunsAQuery() {
             .projectResource(new GoogleProjectResource().googleProjectId("project123"))
             .name("dataset123")
             .id(UUID.randomUUID());
-    TablePointer tablePointer = TablePointer.fromRawSql("table-name");
-    TableVariable tableVariable = TableVariable.forPrimary(tablePointer);
-    Query query =
-        new Query(
-            List.of(
-                new FieldVariable(new FieldPointer(tablePointer, "column-name"), tableVariable)),
-            List.of(tableVariable));
+    var settings = new SnapshotBuilderSettings();
+    when(snapshotBuilderSettingsDao.getSnapshotBuilderSettingsByDatasetId(dataset.getId()))
+        .thenReturn(settings);
+    Query query = mock(Query.class);
     var criteriaQueryBuilderMock = mock(CriteriaQueryBuilder.class);
     when(datasetService.retrieve(dataset.getId())).thenReturn(dataset);
-    when(queryBuilderFactory.criteriaQueryBuilder(any(), any()))
+    when(queryBuilderFactory.criteriaQueryBuilder("person", settings))
         .thenReturn(criteriaQueryBuilderMock);
-    when(criteriaQueryBuilderMock.generateRollupCountsQueryForCriteriaGroupsList(any()))
+    var criteriaGroups = List.of(List.of(new SnapshotBuilderCriteriaGroup()));
+    when(criteriaQueryBuilderMock.generateRollupCountsQueryForCriteriaGroupsList(criteriaGroups))
         .thenReturn(query);
-    when(azureSynapsePdao.runQuery(
-            eq(query.renderSQL(QueryTestUtils.createContext(CloudPlatform.AZURE))), any()))
-        .thenReturn(List.of(5));
+    String sql = "sql";
+    // Not verified: creation of context based on dataset's cloud platform.
```

Comment:
Is this not verified here because its tested in SqlRenderContextTest?

Reply:
Not really, the thing that's not verified here is that a context was created using the platform type in `dataset`. In the code being tested, there's code that creates a context and then passes it to `renderSql()`. The problem is that, since we don't bottleneck that object construction, and since that object returns `live` data, we can't provide an object that will work with an argument matcher.

It occurs to me now that I could write a matcher that tests for the platform type in the context, maybe I'll give that a try.

I think a better unit test would intercept `SnapshotBuilderService.createContext()` so it returns a specific object that we could match the argument exactly. But that would require either changing the way context is constructed, or using a `spy()` of `SnapshotBuilderService`.